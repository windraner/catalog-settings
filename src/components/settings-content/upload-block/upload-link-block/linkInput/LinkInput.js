import React, { Component } from 'react';
import DisabledLinkInput from './disabledLinkInput/DisabledLinkInput';
import PropTypes from 'prop-types';

import './linkInput.css';

export default class LinkInput extends Component {

  changeHandler = (e) => {
    this.props.inputHandler('fileUrl', e.target.value.trim());
  }

  clearHandler = () => {
    this.props.inputHandler('fileUrl', '');
  }

  render() {
    const { placeholder, fileUrl, file } = this.props;

    if(file) {
      return (
        <DisabledLinkInput />
      );
    }

    return (
      <div className="catalog-settings-content__link-input-wrapper">
        <input
          value={fileUrl ? fileUrl : ''}
          onChange={this.changeHandler}
          type="text"
          placeholder={placeholder}
          className="catalog-settings-content__link-input"
        />
        {
          fileUrl ?
            <span
              className="catalog-settings-content__link-input-clear"
              onClick={this.clearHandler}
            />
            :
            null
          }
      </div>
    );
  }
}

LinkInput.propTypes = {
  placeholder: PropTypes.string,
  inputHandler: PropTypes.func.isRequired,
  fileUrl: PropTypes.string.isRequired,
  file: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]).isRequired,
  flags: PropTypes.object,
};
