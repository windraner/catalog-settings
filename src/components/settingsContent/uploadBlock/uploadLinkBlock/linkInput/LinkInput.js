import React, { Component } from 'react';

import './linkInput.css';

export default class LinkInput extends Component {
  render() {
    return (
      <input
        // value={value ? value : ''}
        // onChange={this.changeHandler}
        // onBlur={this.blurHandler}
        type="text"
        className="catalog-settings-content__link-input"
      />
    );
  }
}
