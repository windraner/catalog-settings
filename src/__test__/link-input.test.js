import React from 'react';
import LinkInput from '../components/settings-content/upload-block/upload-link-block/link-input/link-input.js';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<LinkInput />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<LinkInput placeholder="test" inputHandler={function() {}} fileUrl={'fileUrl'} file={''} flags={{}} />);
    const input = wrapper.find('.catalog-settings-content__link-input');
    expect(input.props().placeholder).toBe('test');
    expect(input.props().value).toBe('fileUrl');
    expect(wrapper.find('.catalog-settings-content__link-input-clear').length).toBe(1);
    wrapper.setProps({fileUrl: ''});
    expect(wrapper.find('.catalog-settings-content__link-input-clear').length).toBe(0);
    wrapper.setProps({file: {}});
    expect(wrapper.find('.position-relative').length).toBe(0);
  });
  it('matches the snapshot1', () => {
    const tree = shallow(<LinkInput placeholder="test" inputHandler={function() {}} fileUrl="fileUrl" file={''} flags={{}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot2', () => {
    const tree = shallow(<LinkInput placeholder="test" inputHandler={function() {}} fileUrl="" file={{}} flags={{}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
