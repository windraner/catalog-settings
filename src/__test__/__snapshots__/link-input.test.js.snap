// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkInput /> matches the snapshot1 1`] = `
<div
  className="position-relative"
>
  <input
    className="catalog-settings-content__link-input d-block w-100"
    onChange={[Function]}
    placeholder="test"
    type="text"
    value="fileUrl"
  />
  <span
    className="catalog-settings-content__link-input-clear position-absolute m-auto"
    onClick={[Function]}
  />
</div>
`;

exports[`<LinkInput /> matches the snapshot2 1`] = `<DisabledLinkInput />`;
